File: Course files/Kent - iOS APP進階補充 /20161222 - iOS APP 進階補充 /SwiftHelloWithSwift/HelloObjCWithSwift/ViewController.m
//
//  ViewController.m
//  HelloObjCWithSwift
//
//  Created by Kent Liu on 2015/2/8.
//  Copyright (c) 2015年 Kent Liu. All rights reserved.
//

#import "ViewController.h"

#import "HelloObjCWithSwift-Swift.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    

}

- (void) viewDidAppear:(BOOL)animated {
    
    [super viewDidAppear:animated];
    
    MySwiftClass *swiftObj=[[MySwiftClass alloc] init];
    
    NSString *result = [swiftObj sayHello];
    
    NSDate *resultDate = swiftObj.date;
    
    NSLog(@"%@ - %@",result,[resultDate description]);
    
    // Show message
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:[resultDate description] message:result preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction *ok = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction *action) {
        
        [alert dismissViewControllerAnimated:YES completion:nil];
        
    }];
    
    [alert addAction:ok];
    
    [self presentViewController:alert animated:YES completion:nil];
    
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Course files/Kent - iOS APP進階補充 /20161222 - iOS APP 進階補充 /SwiftHelloWithObjC/HelloWithObjC/ViewController.swift
//
//  ViewController.swift
//  HelloWithObjC
//
//  Created by Kent Liu on 2015/2/8.
//  Copyright (c) 2015年 Kent Liu. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        

    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        
        let objc = MyObjCClass()
        
        let result = objc.sayHello()
        let resultDate = objc.date
        
        let alert = UIAlertController(title: result, message: resultDate?.description, preferredStyle: UIAlertControllerStyle.alert)
        
        let ok = UIAlertAction(title: "OK", style: UIAlertActionStyle.default) { (action) -> Void in
            
            alert.dismiss(animated: true, completion: nil)
        }
        
        alert.addAction(ok)
        
        self.present(alert, animated: true, completion: nil)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

